src/Extensions/Script/Serialization/CustomSerializerJsonConverter.cs

```diff
@@ -0,0 +1,39 @@
+// MIT License.
+
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace System.Web.Script.Serialization;
+
+internal sealed class CustomSerializerJsonConverter(JavaScriptSerializer serializer) : JsonConverter<object>
+{
+    public override object Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+    {
+        throw new Exception("Shouldn't be called.");
+    }
+
+    public override void Write(Utf8JsonWriter writer, object value, JsonSerializerOptions options)
+    {
+        var converter = serializer.GetConverter(value.GetType());
+        var internalObject = converter.Serialize(value, serializer);
+        JsonSerializer.Serialize(writer, internalObject);
+    }
+}
+
+//Factory to check if the type is supported by the JavaScriptSerializer
+internal sealed class CustomSerializerFactory(JavaScriptSerializer serializer) : JsonConverterFactory
+{
+    private readonly JavaScriptSerializer _serializer = serializer;
+    private JavaScriptConverter _converter;
+
+    public override bool CanConvert(Type typeToConvert)
+    {
+        _converter = _serializer.GetConverter(typeToConvert);
```

Comment:
this shouldn't be cached - how are you going to prevent a different call to CanConvert before CreateConverter is called?

Reply:
done.. we actually don't need the field